README.md

```diff
@@ -88,18 +96,53 @@ JUDOSCALE = {
 }
 ```
 
-If you do this, you'll need to retrieve the above dictionary from the flask app's configuration and merge it with Judoscale's configuration. For example, if the flask app's settings are in a form of an object, you can do the following:
+If you do this, you'll need to retrieve the above dictionary from the flask
+app's configuration and merge it with Judoscale's configuration. For example, if
+the flask app's settings are in a form of an object, you can do the following:
 
 ```python
 from judoscale.core.config import config as judoconfig
 
 judoconfig.merge(getattr(config_obj, "JUDOSCALE", {}))
 ```
 
-Note the [official recomendations for configuring Flask](https://flask.palletsprojects.com/en/2.2.x/config/#configuration-best-practices).
+Note the [official recommendations for configuring
+Flask](https://flask.palletsprojects.com/en/2.2.x/config/#configuration-best-practices).
 
 ## Development
 
-This repo inclues a `sample-apps` directory containing apps you can run locally. These apps use the judoscale-python adapter, but they override `API_BASE_URL` so they're not connected to the real Judoscale API. Instead, they post API requests to https://requestcatcher.com so you can observe the API behavior.
+This repo includes a `sample-apps` directory containing apps you can run locally.
+These apps use the `judoscale-python` adapter, but they override `API_BASE_URL` so
+they're not connected to the real Judoscale API. Instead, they post API requests
+to https://requestcatcher.com so you can observe the API behavior.
 
-See the `README` in a sample app for details on how to set it up and run locally.
+See the `README` in a sample app for details on how to set it up and run
+locally.
+
+### Contributing
+
+`judoscale-python` uses [Poetry](https://python-poetry.org/) for managing its
+dependencies and packaging the library.
+
+In order to contribute to `judoscale-python` you should have a recent version
+(1.2+) of Poetry installed.
+
+Clone the repo with
+
+```sh
+$ git clone git@github.com:judoscale/judoscale-python.git
+$ cd judoscale-python
+```
+
+Install dependencies with Poetry and activate the virtualenv
```

Comment:
I think this is missing a step to install Poetry itself:

```sh
judoscale-python $ asdf current
crystal         1.5.0           /Users/adam/.tool-versions
nodejs          17.7.2          /Users/adam/.tool-versions
python          3.10.9          /Users/adam/src/judoscale-python/.tool-versions
ruby            3.0.3           /Users/adam/.tool-versions

judoscale-python $ python --version
Python 3.10.9

judoscale-python $ which poetry
/Users/adam/.asdf/shims/poetry

judoscale-python $ poetry install
No preset version installed for command poetry
Please install a version by running one of the following:

asdf install python 3.10.9

or add one of the following versions in your config file at /Users/adam/src/judoscale-python/.tool-versions
python 3.10.2
```

Looks like there's a [few ways to install Poetry](https://python-poetry.org/docs/), and I don't remember doing any of them when I ran your previous project that used Poetry. I feel like Poetry was already installed. 🤷‍♂️

Reply:
I think this is coming from `asdf`? https://github.com/asdf-vm/asdf/issues/914

FWIW I cannot replicate this behaviour. Also my `poetry` binary is in `/Users/karl/.local/bin/poetry`. I tried uninstalling Poetry completely and re-installing from Homebrew, which also worked without issues. The Homebrew version of Poetry was installed into `/opt/homebrew/bin/poetry`.

What version of Poetry do you have installed?

If I had to guess, I'd say your Poetry installation has somehow ended up somewhere where it shouldn't be. Perhaps you installed Poetry when you were trying out my previous project and the executable somehow ended up in the virtualenv for the that project?

AFAIK, Poetry should be installed "globally". Now, with `asdf`, "globally" is a slightly slippery term, but I don't think `poetry`-the-executable should end up in `~/.asdf/shims`.

I would try uninstalling Poetry completely, then re-installing either according to the [official instructions](https://python-poetry.org/docs/#installation) or from Homebrew.

Reply:
> What version of Poetry do you have installed?

I couldn't tell because I couldn't run the executable. 😳 

But I think I'm good now. I got rid of my `poetry` shim in `~/.asdf/shims` by uninstalling Python 3.10.2. I still don't remember how I originally installed Poetry, but clearly I was using 3.10.2 at the time.

Anyway, now that the shim is gone, I installed Poetry using the [`curl` command on their site](https://python-poetry.org/docs/#installing-with-the-official-installer), and I'm good to go.

```
judoscale-python $ poetry --version
Poetry (version 1.3.1)
```

Thanks!